plugins/autochop.cpp

```diff
@@ -170,38 +189,73 @@ static int string_to_int(string s, int default_ = 0)
     }
 }
 
-static void save_config()
+static void save_config(color_ostream& out)
 {
-    config_autochop.val() = watchedBurrows.getSerialisedIds();
-    config_autochop.ival(0) = autochop_enabled;
-    config_autochop.ival(1) = min_logs;
-    config_autochop.ival(2) = max_logs;
-    config_autochop.ival(3) = wait_for_threshold;
+    //out << "saving data..." << endl;
+    //cerr << __FILE__ << ":" << __LINE__ << endl;
+    Persistent::erase("autochop");
+    Json::Value& node = Persistent::get("autochop");
+    node["version"] = persist_version;
+    node["enabled"] = autochop_enabled;
+    node["min_logs"] = min_logs;
+    node["max_logs"] = max_logs;
+    node["wait_for_threshold"] = wait_for_threshold;
+
+    vector<int32_t> ids;
+    watchedBurrows.getIds(ids);
+
+    Json::Value idsNode;
+    std::for_each(ids.begin(),ids.end(), [&idsNode](int32_t i) { idsNode.append(i); });
+    node["burrow_ids"] = idsNode;
 }
 
-static void initialize()
+static void load_config(color_ostream& out)
 {
+    //call this on_load, regardless of whether the plugin is enabled or not
+    //do NOT call this except on_load
     watchedBurrows.clear();
-    autochop_enabled = false;
+    //default values here
+    //autochop_enabled = false;
+    plugin_self->plugin_enable(out,false);
```

Comment:
A couple things:
- why not just use `plugin_enable(out, false)`?
- does it really need to be disabled?